### PR TITLE
New storybook nav structure for docs; Checkbox, Switch docs

### PR DESCRIPTION
**Pull request by libertymayc**
New nav structure makes it easier to find and ensure that the following is included in all complete component docs:

- API
- Props Table
- HTML Anatomy
- CSS Class
- Characteristics Used
- Custom Property API
- Customising styles

Also changes how stories are referred to by their id rather than re-writing them within the docs. Allows us to also ensure we give supporting documentation to all the existing stories by directly referencing them. I think we should not have each of these listed in the sidebar again, but we should somehow have an 'Examples' anchor in the sidebar just before the first example

### CODE REVIEW

File: packages/core/stories/button.doc.stories.mdx [hunk 1]
```diff
@@ -100,7 +62,9 @@ import { Button } from "@jpmorganchase/uitk-core";
 
 ## Props
 
-<ArgsTable of={Button} />
+<Story name="Props">
```
Comment (joshwooding) [inline packages/core/stories/button.doc.stories.mdx:65]: I understand the benefits but I still don't think we should declare stories in mdx.

File: packages/core/stories/button.doc.stories.mdx [hunk 2]
```diff
@@ -32,61 +26,29 @@ A Button executes an action when clicked. Button labels should use only one or t
 Button contains three variants `cta`, `primary` and `secondary`. If not specified, the default value is `primary`.
 
 <Canvas>
```
Comment (heswell) [inline packages/core/stories/button.doc.stories.mdx:28]: Problem with this approach is that if user view the code , they don't see anything helpful, just a bunch of these:
'''
() => {
  return <ButtonGrid variant="cta" label="CTA" />;
}
'''
We need to make sure the target Story is simpler if we're going to link like this

File: packages/lab/src/pill/Pill.css [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  --pill-text-color-active: var(--uitkPill-text-color-active, var(--uitk-taggable-foreground-active));
+  --pill-text-color-hover: var(--uitkPill-text-color-hover, var(--uitk-taggable-foreground-hover));
+}
+
+/* Button Component API - reset to default state so they don't interfere with Pill styling */
+.uitkPill {
+  --uitkButton-background-active: var(--pill-background);
+  --uitkButton-textColor-active: var(--pill-text-color);
+  --uitkButton-background-hover: var(--pill-background);
+  --uitkButton-textColor-hover: var(--pill-text-color);
```
Comment (heswell) [inline packages/lab/src/pill/Pill.css:37]: I've lost track of what we decided on textColor vs text-color. We seem to have both

File: packages/lab/src/pill/Pill.css [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+/* Style applied to Pill when active if `deletable={false}`, `selectable={false}`, and `disabled={false}` */
+.uitkPill:not(.uitkPill-deletable):not(.uitkPill-selectable):not(.uitkPill-disabled):active,
+.uitkPill:not(.uitkPill-deletable):not(.uitkPill-selectable):not(.uitkPill-disabled).uitkPill-active {
+  background: var(--pill-background-active);
+  color: var(--pill-text-color-active);
+}
+
+/* Styles applied to the Pill checkbox */
+.uitkPill-checkbox {
+  --pill-checkbox-stroke: var(--uitkPill-checkbox-stroke, var(--uitk-selectable-borderColor));
```
Comment (heswell) [inline packages/lab/src/pill/Pill.css:194]: can't see these two variables used anywhere

File: packages/core/stories/button.doc.stories.mdx [hunk 5]
```diff
@@ -111,32 +75,34 @@ export const Link = ({ children, to, ...props }) => (
...
-      elementType={Link}
-      to="/"
-      aria-label="Link as elementType"
-      role={null}
-    >
-      Custom Link
-    </Button>
-  </ComponentAnatomy>
+  <Story name="HTML Anatomy">
+    <ComponentAnatomy showLegend>
```
Comment (heswell) [inline packages/core/stories/button.doc.stories.mdx:79]: Not sure what anybody else thinks, but I've come to see the legend as superfluous - takes up space and adds noise without being very useful. I'd be inclined to drop it everywhere

File: packages/core/stories/button.doc.stories.mdx (hunk 1 above)
Comment (joshwooding) [reply packages/core/stories/button.doc.stories.mdx:65]: To avoid stories showing up in the sidebar.

**Pull request closed without merging**